### PR TITLE
Add env_file + ConfigMaps feature to Kompose

### PR DESCRIPTION
**Pull request by cdrage**
When using env_file with Docker Compose, a ConfigMap will be generated.

For example:

```sh
▶ ./kompose convert -f
script/test/fixtures/configmaps/docker-compose.yml
INFO Kubernetes file "redis-service.yaml" created
INFO Kubernetes file "redis-deployment.yaml" created
INFO Kubernetes file "foo-env-configmap.yaml" created
INFO Kubernetes file "bar-env-configmap.yaml" created
```

File:

```yaml
version: '3'

services:
  redis:
    image: 'bitnami/redis:latest'
    environment:
      - ALLOW_EMPTY_PASSWORD=no
    # Env file will override environment / warn!
    env_file:
      - "foo.env"
      - bar.env
    labels:
      kompose.service.type: nodeport
    ports:
      - '6379:6379'
```

To:

```yaml
apiVersion: v1
data:
  ALLOW_EMPTY_PASSWORD: "yes"
kind: ConfigMap
metadata:
  creationTimestamp: null
  name: foo-env
```

```yaml
...
      - env:
        - name: ALLOW_EMPTY_PASSWORD
          valueFrom:
            configMapKeyRef:
              key: ALLOW_EMPTY_PASSWORD
              name: foo-env
```

### CODE REVIEW

**Comment by cdrage** [comment]
TODO: 
- Add tests
- Remove conflicts (kobject.go)

**Comment by cdrage** [comment]
Ready for review @ashetty1 @containscafeine @kadel @surajnarwade @surajssd

File: pkg/kobject/kobject.go [hunk 1]
```diff
@@ -60,10 +60,10 @@ type ConvertOptions struct {
 
 // ServiceConfig holds the basic struct of a container
 type ServiceConfig struct {
-	// use tags to mark from what element this value comes
 	ContainerName   string
 	Image           string              `compose:"image"`
 	Environment     []EnvVar            `compose:"environment"`
+	EnvFile         []string            `compose:""`
```
Comment (surajnarwade) [inline pkg/kobject/kobject.go]: we can add `compose:"env_file"` as it is valid key

**Comment by surajnarwade** [comment]
version 2 docker -compose file,

```
version: '3'
  
services:
  redis:
    image: 'bitnami/redis:latest'
    environment:
      - ALLOW_EMPTY_PASSWORD=no
    # Env file will override environment / warn!
    env_file:
      - ./bar.env
    labels:
      kompose.service.type: nodeport
    ports:
      - '6379:6379'
```
doesn't create configmaps,

```
$ kompose convert 
INFO Kubernetes file "redis-service.yaml" created 
INFO Kubernetes file "redis-deployment.yaml" created 
```

**Comment by cdrage** [comment]
Your example shows Version 3 @surajnarwade 

Yeah, at the moment it doesn't support Version 2. I will have to add that. Baby steps! :+1: 

Can you do a review on the basis Version 3?

**Comment by surajnarwade** [comment]
@cdrage , my bad about version

:+1:  for baby step

**Comment by cdrage** [comment]
@containscafeine @kadel @surajnarwade @surajssd ready for review :+1:

**Comment by cdrage** [comment]
Reason why this is not in v2 is because of this: https://github.com/docker/libcompose/issues/488

**Comment by containscafeine** [comment]
@cdrage still getting the error in https://github.com/kubernetes/kompose/pull/799#pullrequestreview-59446489 :(

**Comment by cdrage** [comment]
@containscafeine should work now, i've created a new function in order to format the new env name's correctly, see output:

```
github.com/kubernetes/kompose  add-configmap ✗                                                                                                                                                                                                                          19m ⚑ ◒  
▶ ./kompose up
INFO We are going to create Kubernetes Deployments, Services and PersistentVolumeClaims for your Dockerized application. If you need different kind of resources, use the 'kompose convert' and 'kubectl create -f' commands instead. 
 
INFO Deploying application in "default" namespace 
INFO Successfully created Service: redis          
INFO Successfully created Deployment: redis       
INFO Successfully created Config Map: bar-env     

Your application has been deployed to Kubernetes. You can run 'kubectl get deployment,svc,pods,pvc' for details.

github.com/kubernetes/kompose  add-configmap ✗                                                                                                                                                                                                                          19m ⚑ ◒  
▶ kubectl get cm
NAME      DATA      AGE
bar-env   2         5s
```

**Comment by cdrage** [comment]
ignoring vendoring updates, this PR needs review

@kadel @containscafeine @surajnarwade @surajssd please review

File: pkg/transformer/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -158,6 +158,33 @@ func (k *Kubernetes) InitSvc(name string, service kobject.ServiceConfig) *api.Se
...
+	envName := FormatEnvName(envFile)
+
+	// In order to differentiate files, we append to the name and remove '.env' if applicate from the file name
+	configMap := &api.ConfigMap{
+		TypeMeta: unversioned.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name: envName,
```
Comment (surajssd) [inline pkg/transformer/kubernetes/kubernetes.go]: Can this be named like other objects kompose creates? `appName - maifest type.yaml`

Here also we can use same format, just add `appName` before `envName` ?

Comment (cdrage) [reply pkg/transformer/kubernetes/kubernetes.go]: so `appName-configmap-env`?

**Comment by surajnarwade** [comment]
works for me :+1:

File: pkg/transformer/kubernetes/kubernetes.go (hunk 2 above)
Comment (surajssd) [reply pkg/transformer/kubernetes/kubernetes.go]: @cdrage this sounds good!

**Pull request merged**